weldx/tests/test_utility.py

```diff
@@ -498,3 +499,117 @@ def test_xr_time_ref():
 
     da2.weldx.time_ref = t0
     assert da2.time.attrs["time_ref"] == t0
+
+
+def test_non_sense():
+    with pytest.raises(TypeError):
+        ut.compare_nested("asdf", "foo")
+
+    with pytest.raises(TypeError):
+        ut.compare_nested(b"asdf", b"foo")
+
+
+@pytest.mark.parametrize(
+    argnames="a, b, expected",
+    argvalues=[
+        ((1, 2, 3), [1, 2, 3], True),
+        ((1, 2, 3), [1, 2, 0], False),
+        ((1, 2, 3), {"f": 0}, False),
+        ((1, 2, 3), "bar", False),
+        ({"x": [1, 2, 3, 4]}, {"x": [1, 2]}, False),
+        ({"x": [1, 2]}, {"y": [1, 2]}, False),
+    ],
+)
+def test_compare_nested(a, b, expected):
+    assert ut.compare_nested(a, b) == expected
+
+
+@pytest.fixture()
+def _default_dicts():
+    """Return two equivalent more deeply nested structures to be modified by tests."""
+    a = {
+        "foo": np.arange(3),
+        "x": {
+            0: [1, 2, 3],
+        },
+        "bar": True,
+    }
+    b = copy.deepcopy(a)
+    return a, b
+
+
+def test_eq_(_default_dicts):
+    a, b = _default_dicts
+    assert ut.compare_nested(a, b)
+
+
+def test_missing_values(_default_dicts):
+    a, b = _default_dicts
+    b["x"][0].pop(-1)
+    assert not ut.compare_nested(a, b)
+
+
+def test_added_value(_default_dicts):
+    a, b = _default_dicts
+    b["x"][0].append(5)
+    assert not ut.compare_nested(a, b)
+
+
+def test_value_changed(_default_dicts):
+    a, b = _default_dicts
+    b["bar"] = False
+    assert not ut.compare_nested(a, b)
+
+
+def test_key_changed1(_default_dicts):
+    a, b = _default_dicts
+    del b["x"]
+    assert not ut.compare_nested(a, b)
+
+
+def test_key_changed2(_default_dicts):
+    a, b = _default_dicts
+    x = b.pop("x")
+    b["y"] = x
+    assert not ut.compare_nested(a, b)
```

Comment:
Maybe group them all in a class `TestCompareNested`.

Reply:
will do. Thanks.

Reply:
any idea, if there is a good workaround for the need of parameterized methods to be "staticmethod"?

Reply:
ah it simply works with passing self. Sorry for the noise.

Reply:
No, but why is this a problem?

Reply:
When you parameterize tests without marking them static, the first argument gets interpreted as self and collides with the pytest.mark logic. But the parameterize decorator is aware of the self argument (when you give it).

Reply:
I know, just wanted to why not simply make them static unless they need access to the test class ;)

Reply:
a class containing only static methods defeats the purpose of a class IMO (like I did it for this comperator thing here :D). Its also pretty redundant to declare all methods static. If we consider using unittest.TestCase as base, it would make even more sense to have non-static members.

Reply:
I like having test classes for the convenience of grouping tests (and running them all) as well so 👍 for using that